### ISSUE TITLE
[Feature Request] control the number of output/file size if repartitionBeforeWrite is enabled

### ISSUE THREAD

**Issue by isunli** [delta-io/delta#1585]
### Overview

We noticed that `spark.databricks.delta.merge.repartitionBeforeWrite.enabled=true`  will repartition to 1 by default and this can't be configured. It may cause OOM issue if we have a big partition. We want to control the number repartitions (number of files generated) per partition for our table. Or even better, have a configure at file size. 

### Motivation
This will let user control the number of partitions in case of large single file cause OOM error.

### Willingness to contribute

The Delta Lake Community encourages new feature contributions. Would you or another member of your organization be willing to contribute an implementation of this feature?

- [ ] Yes. I can contribute this feature independently.
- [x] Yes. I would be willing to contribute this feature with guidance from the Delta Lake community.
- [ ] No. I cannot contribute this feature at this time.

**Comment by keen85** [comment]
Hi @isunli, your request does sound similar to https://github.com/delta-io/delta/pull/1198.
Will this fit to your requirements?

**Comment by isunli** [comment]
> 

Yes, it is a more powerful version of my request. Please merge this request to #1198

**Comment by foss-contributor** [comment]
@isunli, thank you for your contribution! This issue has been inactive for over 18 months and will be closed in one week. To keep it open, please leave a comment on this issue or contact us in the `#contributors` channel on [Delta Lake Slack](https://go.delta.io/slack). We appreciate your involvement in Delta Lake!
